lib/puppet_x/puppetlabs/strings/yard/handlers/host_class_handler.rb

```diff
@@ -8,6 +8,23 @@ class PuppetX::PuppetLabs::Strings::YARD::Handlers::HostClassHandler < PuppetX::
         param_tuple << ( a[1].nil? ? nil : a[1].source )
       end
     end
+    tp = Puppet::Pops::Types::TypeParser.new
+    param_type_info = {}
+    statement.pops_obj.parameters.each do |pop_param|
+      # If the parameter's type expression is nil, default to Any
+      if pop_param.type_expr == nil
+        param_type_info[pop_param.name] = Puppet::Pops::Types::TypeFactory.any() 
+      else
+        begin
+          param_type_info[pop_param.name] =  tp.interpret_any(pop_param.type_expr)
+        rescue Puppet::ParseError
+          # If the type could not be interpreted, default to type Any
+          param_type_info[pop_param.name] = "TypeError - " +
+            "#{pop_param.type_expr} isn't a valid type."
```

Comment:
On further inspection, interpret_any can't even raise this exception. The interpret method can, and I originally thought that interpret_any called out to interpret. I have looked into what other exceptions it can raise (is there a tool to make this easier?) but nothing seems likely. This is ruby, so potentially anything can be passed down this hole, but I can't defend against absolutely every exception. I'm going to remove this code path altogether.